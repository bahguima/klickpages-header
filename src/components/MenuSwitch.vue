<template>
  <div class="menu--switch">
    <ul class="container no-padding">
      <li>
        <i class="icon-kp-kp"></i>
        Klickpages
      </li>
      <li class="myKlicksendMenu">
        <a :href="klicksendUrl" data-turbolinks="false">
          <i class="icon-kp-ks"></i>
          Klicksend
        </a>
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    klicksendUrl: String,
  },
};
</script>
