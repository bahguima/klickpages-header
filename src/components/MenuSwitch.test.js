import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import MenuSwitch from './MenuSwitch.vue';

describe('components/MenuSwitch.vue', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  let wrapper;

  const topBar = {
    namespaced: true,
  };

  describe('when component is mounted', () => {
    describe('and klicksend topBarConfig is enabled', () => {
      beforeAll(() => {
        topBar.getters = {
          config: () => ({
            klicksend: {
              enabled: true,
            },
          }),
        };

        const store = new Vuex.Store({ modules: { topBar } });
        wrapper = shallowMount(MenuSwitch, { localVue, store });
      });

      it('should renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
      });

      it('should contains menu--switch class', () => {
        expect(wrapper.contains('.menu--switch')).toBe(true);
      });

      afterAll(() => {
        wrapper.destroy();
        jest.clearAllMocks();
      });
    });

    describe('and klicksend topBarConfig is not enabled', () => {
      beforeAll(() => {
        topBar.getters = {
          config: () => ({
            klicksend: {
              enabled: false,
            },
          }),
        };

        const store = new Vuex.Store({ modules: { topBar } });
        wrapper = shallowMount(MenuSwitch, { localVue, store });
      });

      it('should renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
      });

      afterAll(() => {
        wrapper.destroy();
        jest.clearAllMocks();
      });
    });
  });
});
