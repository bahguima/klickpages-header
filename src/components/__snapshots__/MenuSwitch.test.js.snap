// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MenuSwitch.vue when component is mounted and klicksend topBarConfig is enabled should renders correctly 1`] = `
<div
  class="menu--switch"
>
  <ul
    class="container no-padding"
  >
    <li>
      <i
        class="icon-kp-kp"
      />
      
      Klickpages
    
    </li>
     
    <li
      class="myKlicksendMenu"
    >
      <a
        data-turbolinks="false"
      >
        <i
          class="icon-kp-ks"
        />
        
        Klicksend
      
      </a>
    </li>
  </ul>
</div>
`;

exports[`components/MenuSwitch.vue when component is mounted and klicksend topBarConfig is not enabled should renders correctly 1`] = `<!---->`;
